### PR TITLE
Move  execution listener to a separate module

### LINKED ISSUE #997: Move TogglzTestExecutionListener out of togglz-spring module
The [TogglzTestExecutionListener](https://github.com/togglz/togglz/blob/master/spring-core/src/main/java/org/togglz/spring/test/TogglzTestExecutionListener.java) will be enabled by default for tests, and it is better to move it standalone module and allow developers to add it manually.

Additionally when I tried to mock `FeatureManager` in my tests it always failed my mocks.

### PR DESCRIPTION
**Pull request by hantsy**
fixes #997

### CODE REVIEW

**Comment by hantsy** [comment]
@bennetelli I tried to add a test for the test execution listener, but I am not sure I could add more assertions to verify the operation in the `TogglzTestExecutionListener`.

**Review by bennetelli — CHANGES_REQUESTED** [review]

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -28,6 +28,6 @@ jobs:
           distribution: 'temurin'
           java-version: ${{ matrix.jdk }}
       - name: Build with Maven
-        run: mvn -P wildfly-managed --batch-mode verify
+        run: mvn -B verify -Pwildfly-managed
```
[the remark this answers is not in the document]
Comment (bennetelli) [reply .github/workflows/maven.yml:31]: ah thanks! I thought it would be an option provided by the wildfly maven plugin. Thanks for the explanation

**Review by bennetelli — CHANGES_REQUESTED** [review]
I thought about this once again. IMO it would be better to integrate it in the existing spring boot starters instead of having to add the starter and the spring-test dependency to every project.

In the end there should be the spring-boot-starter only if it is possible.

**Comment by bennetelli** [comment]
I think we need to find a better way to achieve this. With these changes, projects using togglz with Feature enum would not need to add the new spring-test dependency, but projects using the application.yml file to declare their feature toggles would need to do it. 

IMO that should be hidden for the users and they should not have to add an additional dependency in this case.

**Comment by hantsy** [comment]
We gave up this in projects.

**Pull request closed without merging**